pkg/acquisition/modules/file/file.go

```diff
@@ -0,0 +1,363 @@
+		}
+
+	}
+	return nil
+}
+
+func (f *FileSource) GetMetrics() []prometheus.Collector {
+	return []prometheus.Collector{linesRead}
+}
+
+func (f *FileSource) GetName() string {
+	return "file"
+}
+
+func (f *FileSource) CanRun() error {
+	return nil
+}
+
+func (f *FileSource) StreamingAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	f.logger.Debug("Starting live acquisition")
+	t.Go(func() error {
+		return f.monitorNewFiles(out, t)
+	})
+	for _, file := range f.files {
+		tail, err := tail.TailFile(file, tail.Config{ReOpen: true, Follow: true, Poll: true, Location: &tail.SeekInfo{Offset: 0, Whence: io.SeekEnd}})
+		if err != nil {
+			f.logger.Errorf("Could not start tailing file %s : %s", file, err)
+			continue
+		}
+		f.tails[file] = true
+		t.Go(func() error {
+			defer types.CatchPanic("crowdsec/acquis/file/live/fsnotify")
+			return f.tailFile(out, t, tail)
+		})
+	}
+	return nil
+}
+
+func (f *FileSource) Dump() interface{} {
+	return f
+}
+
+func (f *FileSource) monitorNewFiles(out chan types.Event, t *tomb.Tomb) error {
+	for {
+		select {
+		case event, ok := <-f.watcher.Events:
+			if !ok {
+				return nil
+			}
+			if event.Op&fsnotify.Create == fsnotify.Create {
+				fi, err := os.Stat(event.Name)
+				if err != nil {
+					f.logger.Errorf("Could not stat() new file %s, ignoring it : %s", event.Name, err)
+					continue
+				}
+				if fi.IsDir() {
+					continue
+				}
+				f.logger.Infof("Detected new file %s", event.Name)
+				matched := false
+				for _, pattern := range f.config.Filenames {
+					f.logger.Debugf("Matching %s with %s", pattern, event.Name)
+					matched, err = path.Match(pattern, event.Name)
+					if err != nil {
+						f.logger.Errorf("Could not match pattern : %s", err)
+						continue
+					}
+					if matched {
+						break
+					}
+				}
+				if !matched {
+					continue
+				}
+				if f.tails[event.Name] {
+					//we already have a tail on it, do not start a new one
+					f.logger.Debugf("Already tailing file %s, not creating a new tail", event.Name)
+					break
+				}
+				//Slightly different parameters for Location, as we want to read the first lines of the newly created file
+				tail, err := tail.TailFile(event.Name, tail.Config{ReOpen: true, Follow: true, Poll: true, Location: &tail.SeekInfo{Offset: 0, Whence: io.SeekStart}})
+				if err != nil {
+					f.logger.Errorf("Could not start tailing file %s : %s", event.Name, err)
+					break
+				}
+				f.tails[event.Name] = true
+				t.Go(func() error {
+					defer types.CatchPanic("crowdsec/acquis/tailfile")
+					return f.tailFile(out, t, tail)
+				})
+			}
+		case err, ok := <-f.watcher.Errors:
+			if !ok {
+				return nil
+			}
+			f.logger.Errorf("Error while monitoring folder: %s", err)
+		}
+	}
+}
+
+func (f *FileSource) tailFile(out chan types.Event, t *tomb.Tomb, tail *tail.Tail) error {
+	//lint:ignore SA1015 as it is an infinite loop
+	timeout := time.Tick(1 * time.Second)
+	f.logger.Debugf("-> Starting tail of %s", tail.Filename)
```

Comment:
Don't forget to fix this, even if the function is endless, this might be ended and restarted in case of SIGHUP

Reply:
Will fix.